main.tf

```
@@ -192,7 +192,7 @@ resource "aws_cloudwatch_log_group" "this" {
 ################################################################################
 
 resource "aws_appautoscaling_target" "this" {
-  count = var.create ? 1 : 0
+  count = var.create && var.create_appautoscaling_target ? 1 : 0
```

lets just create one variable that is used to enable/disable autoscaling
```suggestion
  count = var.create && var.enable_storage_autoscaling ? 1 : 0
```